share/completions/killall.fish

```diff
@@ -4,17 +4,48 @@ for i in $__kill_signals
 	set number (echo $i | cut -d " " -f 1)
 	set name (echo $i | cut -d " " -f 2)
 	complete -c killall -o $number -d $name
-	complete -c killall -o $name -d $name
-	complete -c killall -o s -x -a \"$number\tSend\ $name\ signal\"
-	complete -c killall -o s -x -a \"$name\tSend\ $name\ signal\"
+	complete -c killall -o $name
+	# Doesn't work in OS X; -s is simulate
+	test (uname) != 'Darwin'; and complete -c killall -s s -x -a "$number $name"
+end
+
+# Finds and completes all users, and their respective UID.
+function __make_users_completions
+	set -l users (dscl . list /Users | grep -v '^_.*')
+	for user in $users
+		set -l uid (id -u $user)
+		# GNU doesn't support UID
+		killall --version > /dev/null ^ /dev/null; and set -el uid
+		complete -c killall -s u -x -a "$user $uid" -d "Match only processes belonging to $user"
+	end
 end
 
 complete -c killall -xa '(__fish_complete_proc)'
-complete -c killall -s e -l exact -d 'Require an exact match for very long names'
-complete -c killall -s I -l ignore-case -d 'Do case insensitive process name match'
-complete -c killall -s g -l process-group -d 'Kill the process group to which the process belongs. The kill signal is only sent once per group, even if multiple processes belonging to the same process group were found'
-complete -c killall -s i -l interactive -d 'Interactively ask for confirmation before killing'
-complete -c killall -s u -l user -d 'Kill only processes the specified user owns. Command names are optional'
-complete -c killall -s w -l wait -d 'Wait  for  all  killed processes to die. killall checks once per second if any of the killed processes still exist and only returns if none are left.  Note that killall may wait forever if the signal was ignored, had no effect, or if the process stays in zombie state'
+
+if killall --verison > /dev/null ^ /dev/null
```

Comment:
> Also, @zanchey mentioned in #2587 that calling killall on Solaris wasn't safe (there, killall just kills all processes on the system, so you don't want to execute it or it executes you).

What's the output of `uname`for Solaris? And how would you stop user from executing that, what if they do want to intentionally do that? Perhaps if the system is solaris, then define a function with name `killall` that requires confirmation?